diff --git a/lib/agents.js b/lib/agents.js
index 45ec59c4c13757379095131c4f0a5ea6f7284f45..0763b031e355a755ec6a26f98461aa3f63b8339b 100644
--- a/lib/agents.js
+++ b/lib/agents.js
@@ -32,7 +32,7 @@ module.exports = class Agent extends AgentBase {
   }
 
   get proxy () {
-    return this.#proxy ? { url: this.#proxy } : {}
+    return this.#proxy ? { url: this.#proxy.toString() } : {}
   }
 
   #getProxy (options) {
